### PR TITLE
Use S3 for fastlane match

### PR DESCRIPTION
**Pull request by crazytonyli**
What says in the title.

You can verify this change by running `bundle exec fastlane ios alpha_code_signing && bundle exec fastlane ios appstore_code_signing` locally.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/Matchfile [hunk 1]
```diff
@@ -3,7 +3,5 @@
 # This Matchfile has the shared properties used for all signing types
 
 # Store certs/profiles encrypted in Google Cloud
-storage_mode('google_cloud')
-google_cloud_bucket_name('a8c-fastlane-match')
-secrets_directory = File.join(Dir.home, '.configure', 'woocommerce-ios', 'secrets')
-google_cloud_keys_file(File.join(secrets_directory, 'google_cloud_keys.json'))
+storage_mode('s3')
+s3_bucket('a8c-fastlane-match')
```
Comment (mokagio) [inline fastlane/Matchfile:7]: We'll need to add `s3_region` to address a runtime warning. See https://github.com/Automattic/simplenote-macos/pull/1150#discussion_r1591913764

```suggestion
s3_bucket('a8c-fastlane-match')
s3_region('us-east-2')
```

**Review by mokagio — APPROVED** [review]
What do you think of taking the chance to:

- Remove `Matchfile`
- Use `app_store_connect_api_key` like [in Pocket Casts Desktop](https://github.com/Automattic/pocket-casts-desktop/pull/46/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688R29-R33)

Of course, those are "refactors" as they don't change the behavior. Approving to unblock.

**Comment by crazytonyli** [comment]
@mokagio Thanks for the review. I'll focus on the fastlane match repo migration for now 🙂

**Pull request merged**